### PR TITLE
Fix some warnings in div_grad_impl.cu

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

The fix is based on my very limited knowledge of this file. I don't understand most part of the code.  I even don't think my fix is quite right. But it is better than nothing. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu [hunk 1]
```diff
@@ -478,7 +478,7 @@ void ImplDivGrad(
           *fdm_output_strides,
           db_output_data,
           N);
-  } else if (a_padded_strides && a_padded_strides->Size()) {
+  } else if (a_padded_strides && b_padded_strides && a_padded_strides->Size()) {
```
Comment (snnn) [inline orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu:481]: avoid deferencing b_padded_strides if b_padded_strides is NULL.

Comment (snnn) [reply orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu:481]: I don't quite understand the code ....

Comment (snnn) [reply orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu:481]: @edgchen1 , would you like to take over it?

**Review by liqunfu — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]

File: orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu (hunk 1 above)
Comment (edgchen1) [reply orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu:481]: ok sure

**Comment by snnn** [comment]
@edgchen1 will create a new PR.

File: orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu (hunk 1 above)
Comment (snnn) [reply orttraining/orttraining/training_ops/cuda/math/div_grad_impl.cu:481]: Thank you!

**Pull request closed without merging**